Controller/CRUDController.php

```diff
@@ -465,7 +465,17 @@ public function batchAction()
         if (count($idx) > 0) {
             $this->admin->getModelManager()->addIdentifiersToQuery($this->admin->getClass(), $query, $idx);
         } elseif (!$allElements) {
-            $query = null;
+            $this->addFlash(
+                'sonata_flash_info',
+                $this->trans('flash_batch_no_elements_processed', [], 'SonataAdminBundle')
+            );
+
+            return new RedirectResponse(
```

Comment:
This should be changed after merging https://github.com/sonata-project/SonataAdminBundle/pull/4741